celery/app/amqp.py

```diff
@@ -589,6 +590,15 @@ def router(self):
     def router(self, value):
         return value
 
+    @cached_property
+    def repr(self):
+        repr_function = self.app.conf.task_args_repr_function
+        if not repr_function:
```

Comment:
In my opinion, this fallback is inappropriate for most if not all cases.
We should raise an `ImproperlyConfigured` exception instead if that is the case.

Reply:
With task_protocol=1, the backend receives the original `args` and `kwargs` values (i.e. as `tuple` and `dict`)
With this fallback I was trying to get the same.
Anyway, I will just raise `ImproperlyConfigured`, explicit is better than implicit.